src/authenticators/mod.rs

```diff
@@ -75,12 +86,35 @@ impl std::fmt::Display for ApplicationName {
 #[serde(tag = "auth_type")]
 pub enum AuthenticatorConfig {
     /// Direct authentication
-    Direct,
-    /// Unix Peer Credenditals authentication
-    UnixPeerCredentials,
+    Direct {
+        /// List of service admins
+        admins: Option<Vec<Admin>>,
+    },
+    /// Unix Peer Credentials authentication
+    UnixPeerCredentials {
+        /// List of service admins
+        admins: Option<Vec<Admin>>,
+    },
     /// JWT-SVID
     JwtSvid {
         /// Path to the Workload API socket
         workload_endpoint: String,
+        /// List of service admins
+        admins: Option<Vec<Admin>>,
     },
 }
+
+/// Structure defining the properties of a service admin
+#[derive(Deserialize, Debug, Zeroize, Clone)]
+#[zeroize(drop)]
+pub struct Admin {
+    name: String,
+}
+
+impl Admin {
+    fn check_app_name(&self, app_name: &mut ApplicationName) {
```

Comment:
Similarly - wasn't sure what name to give to this, but it felt better to keep the place where the actual check is done and `set_admin` is called contained, so it's easier to verify